cmd/kubelet/app/server.go

```diff
@@ -202,12 +202,20 @@ HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 				if err := kubeletConfigFlagPrecedence(kubeletConfig, args); err != nil {
 					klog.Fatal(err)
 				}
+
 				// update feature gates based on new config
 				if err := utilfeature.DefaultMutableFeatureGate.SetFromMap(kubeletConfig.FeatureGates); err != nil {
 					klog.Fatal(err)
 				}
 			}
 
+			// For backward compatibility, if EnableSystemLogHandler is not specified in the config copy the value
+			// from EnableDebuggingHandlers. This can eventually move into v1beta1/defaults.go once
+			// --enable-debugging-handlers flag is removed. see https://github.com/kubernetes/kubernetes/pull/87273
+			if kubeletConfig.EnableSystemLogHandler == nil {
+				kubeletConfig.EnableSystemLogHandler = &kubeletConfig.EnableDebuggingHandlers
```

Comment:
this is done before the dynamic config is loaded (the `if dynamicConfigDir` block below)... doesn't this mean we will not honor a EnableDebuggingHandlers set via dynamic config?

Reply:
Ah my bad, I missed this. Good catch, I will fix this. More I think about this, I feel like there has to be single method which sets default value and then invokes ValidateKubeletConfiguration instead of setting defaults differently for each cases (Config/CLI) and invoking ValidateKubeletConfiguration in different places. I can own cleaning this after I'm done with this PR.

Given that ValidateKubeletConfiguration is invoked in three places, unfortunately adding a check enableSystemLogHandler would fail when ValidateKubeletConfiguration is invoked right after reading the flag values and also after reading the configMap (associated with dynamicConfig) since the enableSystemLogHandler will not be set.

I'm adding a check for now to verify that enableSystemLogHandler is set before starting the kubelet. So that if some new consumers forget to update the defaults before invoking runKubelet it will fail properly.

Reply:
Pushed a new commit https://github.com/kubernetes/kubernetes/pull/87273/commits/e70e2d1561a8e3e1d7311765cde1932598619f40 addressing this. Thanks again for taking time in reviewing all these changes.